File: capstone_project/Backend/flightms/flightms/src/main/java/com/wipro/flightms/service/FlightService.java
package com.wipro.flightms.service;

import com.wipro.flightms.dto.response.AirlineResponse;
import com.wipro.flightms.dto.response.FlightResponse;

import java.time.LocalDate;
import java.util.List;

public interface FlightService {
    public List<FlightResponse> getAllFlights();
    public List<FlightResponse> getFlightsBySourceAndDestination(String source, String destination, LocalDate time);
    public FlightResponse getFlightById(Long id);
    public List<AirlineResponse> getAllAirlines();
}

File: capstone_project/Frontend/FlightBookingApp/src/app/services/flight.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import {
  FlightResponse,
  AirlineResponse,
  FlightSearchRequest
} from '../models/flight.model';

@Injectable({
  providedIn: 'root',
})
export class FlightService {
  private apiUrl = 'http://localhost:9191/flight-ms/api'; 

  constructor(private http: HttpClient) {}

  getAllFlights(): Observable<FlightResponse[]> {
    return this.http.get<FlightResponse[]>(`${this.apiUrl}/all-flights`);
  }

  getFlightsBySourceAndDestination(
    sourceCode: string,
    destinationCode: string,
    date: string
  ): Observable<FlightResponse[]> {
    const params = new HttpParams()
      .set('source', sourceCode)
      .set('destination', destinationCode)
      .set('date', date);

    return this.http.get<FlightResponse[]>(`${this.apiUrl}/get-flights`, {
      params,
    });
  }

  getFlightById(id: number): Observable<FlightResponse> {
    return this.http.get<FlightResponse>(`${this.apiUrl}/flights/${id}`);
  }

  getAllAirlines(): Observable<AirlineResponse[]> {
    return this.http.get<AirlineResponse[]>(`${this.apiUrl}/airlines`);
  }

}
